### PR TITLE
Allows item specials to smack bots (secbots/medibots/etc)

### PR DESCRIPTION
**Pull request by frawhst**
[feature]
## About the PR 
Does that.

## Why's this needed? 
Fighting emagged/evil medibots can be annoying because special attacks don't work. Ditto for secbots.

```changelog
(u)aloe
(+)Special attacks can now be used to bonk securitrons, medibots, and other small robots.
```

### CODE REVIEW

**Review by Tarmunora — CHANGES_REQUESTED** [review]
Items specials previously could safely assume that any valid targets would be of type `mob/living`.
Please ensure that existing itemspecials will not runtime when used on a bot

**Comment by frawhst** [comment]
> Items specials previously could safely assume that any valid targets would be of type `mob/living`.
> Please ensure that existing itemspecials will not runtime when used on a bot

Item specials also already accept `/obj/critter` as valid targets, which... already causes runtimes in testing (baton special, probably others). I'll go through and fix those as I find them.

**Comment by Tarmunora** [comment]
> 
> 
> > Items specials previously could safely assume that any valid targets would be of type `mob/living`.
> > Please ensure that existing itemspecials will not runtime when used on a bot
> 
> Item specials also already accept `/obj/critter` as valid targets, which... already causes runtimes in testing (baton special, probably others). I'll go through and fix those as I find them.

lol rip

**Comment by frawhst** [comment]
spark was the only one I found that consistently broke, everything else should work fine.

**Review by Tarmunora — APPROVED** [review]

File: code/WorkInProgress/ItemSpecials.dm [hunk 1]
```diff
@@ -1085,8 +1085,19 @@
...
+				M.bodytemperature += 4 * mult
+			else if (iscritter(hit))
+				var/obj/critter/C = hit
+				C.health -= 10
+				C.check_health()
+			else //bot
+				var/obj/machinery/bot/B = hit
+				B.health -= 10
+				if (B.health <= 0)
+					B.explode()
```
Comment (Sovexe) [inline code/WorkInProgress/ItemSpecials.dm:1100]: this is not a safe way to handle damage. 
* This bypasses other procs to wake up hibernating critters up and is exploitable
* This bypasses critter vulnerabilities and defenses
* You should not be handling deaths in the weapon special

File: code/WorkInProgress/ItemSpecials.dm [hunk 2]
```diff
@@ -1409,14 +1420,25 @@
...
+					M.bodytemperature += 4 * mult
+				else if (iscritter(hit))
+					var/obj/critter/C = hit
+					C.health -= 10
+					C.check_health()
+				else //bot
+					var/obj/machinery/bot/B = hit
+					B.health -= 10
+					if (B.health <= 0)
+						B.explode()
```
Comment (Sovexe) [inline code/WorkInProgress/ItemSpecials.dm:1441]: ditto

**Review by Sovexe — CHANGES_REQUESTED** [review]

**Comment by frawhst** [comment]
Removed the effects on the spark special from critters/bots- only affects mobs now. Basically, same behavior as before, but no runtime.

**Comment by Sovexe** [comment]
having it it the bots and critters is fine, but you would probably want to either make something like `special_act` or just some sort of generic take damage equivalent for obj critters and bots like mobs are using

**Review by Tarmunora — APPROVED** [review]

**Comment by frawhst** [comment]
I'd be open to creating generic `TakeDamage` procs for critters/bots, but there's a bunch of places those should probably be used (fireflashes, etc), and I'd rather not do all of that in this PR, considering that would get way out of scope. Something for later.

File: code/WorkInProgress/ItemSpecials.dm [hunk 3]
```diff
@@ -1409,16 +1410,17 @@
...
 			// 	H.do_disorient(src.stamina_damage * mult, weakened = 10)
 			if(istype(master, /obj/item))
-				hit.TakeDamage("chest", 0/*master.force*/, rand(2 * mult,5 * mult), 0, DAMAGE_BLUNT)
-				hit.bodytemperature += 4 * mult
-
-			playsound(hit, 'sound/effects/electric_shock.ogg', 60, 1, 0.1, 2.8)
+				if (ismob(hit))
+					var/mob/M = hit
+					M.TakeDamage("chest", 0, rand(2 * mult,5 * mult), 0, DAMAGE_BLUNT)
+					M.bodytemperature += 4 * mult
```
Comment (Sovexe) [inline code/WorkInProgress/ItemSpecials.dm:1422]:
```suggestion
					M.bodytemperature += (4 * mult)
```

File: code/WorkInProgress/ItemSpecials.dm [hunk 4]
```diff
@@ -1080,15 +1080,16 @@
...
 				H.do_disorient(src.stamina_damage * mult, weakened = 10)
 
-			hit.TakeDamage("chest", 0, rand(2 * mult,5 * mult), 0, DAMAGE_BLUNT)
-			hit.bodytemperature += 4 * mult
-
-			playsound(hit, 'sound/effects/electric_shock.ogg', 60, 1, 0.1, 2.8)
+			if (ismob(hit))
+				var/mob/M = hit
+				M.TakeDamage("chest", 0, rand(2 * mult, 5 * mult), 0, DAMAGE_BLUNT)
+				M.bodytemperature += 4 * mult
```
Comment (Sovexe) [inline code/WorkInProgress/ItemSpecials.dm:1091]:
```suggestion
				M.bodytemperature += (4 * mult)
```

**Review by Sovexe — APPROVED** [review]

**Pull request merged**